app/controllers/application_controller.rb

```diff
@@ -27,6 +27,38 @@ class ApplicationController < ActionController::Base
 
   helper_method :current_person
 
+  helper_method def url_for(options)
+    space = if options[0].is_a?(Space) && options[0].branded_domain.present?
+      options.delete_at(0)
+    elsif [:edit, :new].include?(options[0]) && options[1].is_a?(Space) && options[1].branded_domain.present?
+      options.delete_at(1)
+    elsif options.is_a?(Space) && options.branded_domain.present?
```

Comment:
Not sure if it is worth it, or if it would make the code even harder to understand, but these checks for "is this a Space and does it have a branded domain" could be tightened up like:
```ruby
thing_were_checking.try(:branded_domain).present?
```
Which is a little bit more flexible (will work for any object that has a `branded_domain`, even if it is not a `Space`), but I think works for our purposes.

Reply:
I always forget about `.try(:foo)`!